pkg/kubelet/kubelet.go

```diff
@@ -2780,14 +2780,49 @@ func (kl *Kubelet) HandlePodUpdates(pods []*v1.Pod) {
 	}
 }
 
-// recordContainerResizeOperations records if any of the pod's containers needs to be resized, and returns
+// recordResizeOperaations records if any of the pod level resources or
+// containers need to be resized, and returns
 // true if so
-func recordContainerResizeOperations(oldPod, newPod *v1.Pod) bool {
-	hasResize := false
+func recordResizeOperations(oldPod, newPod *v1.Pod) bool {
 	if oldPod == nil {
 		// This should never happen.
 		return true
 	}
+
+	hasResize := false
+	hasResize = recordPodLevelResourceResizeOperations(oldPod, newPod)
+	hasResize = recordContainerResizeOperations(oldPod, newPod)
+	return hasResize
```

Comment:
You are unconditionally overwriting the pod-level hasResize with the container-level hasResize, making the first one ineffectual

```suggestion
	return recordPodLevelResourceResizeOperations(oldPod, newPod) || recordContainerResizeOperations(oldPod, newPod)
```